Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserViewUtils.cpp

```diff
@@ -510,5 +515,111 @@ namespace AzToolsFramework
+                {
+                    static constexpr const char* FolderIconPath = "Assets/Editor/Icons/AssetBrowser/GemFolder_80.svg";
+                    iconPathToUse = (engineRoot / FolderIconPath).c_str();
+                }
+                else
+                {
+                    static constexpr const char* FolderIconPath = "Assets/Editor/Icons/AssetBrowser/Folder_80.svg";
+                    iconPathToUse = (engineRoot / FolderIconPath).c_str();
+                }
+                return iconPathToUse;
+            }
+
+            // Check if this entry has a custom previewer, if so use that thumbnail
+            AZ::EBusAggregateResults<const PreviewerFactory*> factories;
+            PreviewerRequestBus::BroadcastResult(factories, &PreviewerRequests::GetPreviewerFactory, entry);
+            for (const auto factory : factories.values)
+            {
+                if (factory)
+                {
+                    SharedThumbnail thumbnail;
+
+                    ThumbnailerRequestBus::BroadcastResult(
+                        thumbnail, &ThumbnailerRequests::GetThumbnail, entry->GetThumbnailKey());
+                    AZ_Assert(thumbnail, "The shared thumbnail was not available from the ThumbnailerRequestBus.");
+                    if (thumbnail && thumbnail->GetState() != Thumbnail::State::Failed)
+                    {
+                        return thumbnail->GetPixmap();
+                    }
+                }
+            }
+            // Helper function to find the full path for a given icon
+            auto findIconPath = [](QString resultPath)
+            {
+                // is it an embedded resource or absolute path?
+                bool isUsablePath =
+                    (resultPath.startsWith(":") || (!AzFramework::StringFunc::Path::IsRelative(resultPath.toUtf8().constData())));
+
+                if (!isUsablePath)
+                {
+                    // getting here means it needs resolution.  Can we find the real path of the file?  This also searches in gems
+                    // for sources.
+                    bool foundIt = false;
+                    AZStd::string watchFolder;
+                    AZ::Data::AssetInfo assetInfo;
+                    AssetSystemRequestBus::BroadcastResult(
+                        foundIt,
+                        &AssetSystemRequestBus::Events::GetSourceInfoBySourcePath,
+                        resultPath.toUtf8().constData(),
+                        assetInfo,
+                        watchFolder);
+
+                    if (foundIt)
+                    {
+                        // the absolute path is join(watchfolder, relativepath); // since its relative to the watch folder.
+                        AZStd::string finalPath;
+                        AzFramework::StringFunc::Path::Join(watchFolder.c_str(), assetInfo.m_relativePath.c_str(), finalPath);
+                        resultPath = QString::fromUtf8(finalPath.c_str());
```

Comment:
This doesn't have to use StringFunc at all, it can use QDir directly to skip over the AZ code.

FYI: The StringFunc::Path functions are are deprecated and should be avoided
Using either AZ::IO::Path* or QDir functions when working with QT should be used instead
```suggestion
                        // the absolute path is join(watchfolder, relativepath); // since its relative to the watch folder.
                        resultPath = QDir(watchFolder.c_str()).absoluteFilePath(assetInfo.m_relativePath.c_str());
```

Reply:
Sounds good, removed and implemented the new change